apps/condo/domains/user/schema/ConfirmPhoneAction.js

```diff
@@ -0,0 +1,245 @@
+        update: access.canManageConfirmPhoneActions,
+        delete: false,
+        auth: true,
+    },
+})
+
+
+const ConfirmPhoneActionService = new GQLCustomSchema('ConfirmPhoneActionService', {
+    queries: [
+        { 
+            access: true,
+            schema: 'getPhoneByConfirmPhoneActionToken(token: String!): String',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { token } = args
+                const now = extra.extraNow || Date.now()
+                const actions = await ConfirmPhoneActionGQL.getAll(context.createContext({ skipAccessControl: true }), { 
+                    token,
+                    expiresAt_gte: new Date(now).toISOString(),
+                    completedAt: null,
+                })
+                if (isEmpty(actions)) {
+                    throw new Error(`${CONFIRM_PHONE_ACTION_EXPIRED}]: Unable to find confirm phone action by token`)
+                }
+                const { phone } = actions[0]
+                return phone
+            },
+        },
+    ],    
+    mutations: [
+        {
+            access: true,
+            schema: 'startConfirmPhoneAction(phone: String!, dv:Int!, sender: JSON!): String',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { phone: inputPhone, sender, dv } = args
+                const phone = normalizePhone(inputPhone)
+                const token = uuid()
+                const now = extra.extraNow || Date.now()
+                const requestedAt = new Date(now).toISOString()
+                const expiresAt = new Date(now + CONFIRM_PHONE_ACTION_EXPIRY * 1000).toISOString()
+                const smsCode = generateSmsCode(phone)
+                const smsCodeRequestedAt = new Date(now).toISOString()
+                const smsCodeExpiresAt = new Date(now + SMS_CODE_TTL * 1000).toISOString()
+                const variables = { 
+                    dv,
+                    sender,
+                    phone,
+                    smsCode,
+                    token,
+                    smsCodeRequestedAt,
+                    smsCodeExpiresAt,
+                    requestedAt,
+                    expiresAt,
+                }
+                await ConfirmPhoneActionGQL.create(context.createContext({ skipAccessControl: true }), variables)
+                const lang = COUNTRIES[RUSSIA_COUNTRY].locale
+                await sendMessage(context, {
+                    lang,
+                    to: { phone },
+                    type: SMS_VERIFY_CODE_MESSAGE_TYPE,
+                    meta: {
+                        dv: 1,
+                        smsCode,                        
+                    },
+                    sender: sender,
+                })
+                return token
+            },
+        },
+        {
+            access: true,
+            schema: 'confirmPhoneActionResendSms(token: String!, sender: JSON!): String',
```

Comment:
it's better to use `confirmPhoneActionResendSms(data: ConfirmPhoneActionResendSmsInput!): ConfirmPhoneActionResendSmsOutput` to extend the return data